authority/authority.go

```diff
@@ -0,0 +1,69 @@
+package authority
+
+import (
+	"time"
+
+	"github.com/go-phorce/dolly/xlog"
+	"github.com/go-phorce/dolly/xpki/cryptoprov"
+	"github.com/go-phorce/trusty/config"
+	"github.com/juju/errors"
+)
+
+var logger = xlog.NewPackageLogger("github.com/go-phorce/trusty", "authority")
+
+// Authority defines the CA
+type Authority struct {
+	issuers          map[string]*Issuer // label => Issuer
+	issuersByProfile map[string]*Issuer // cert profile => Issuer
+
+	caConfig Config
```

Comment:
just curious, why does not each issuer has its own caConfig. For example, each issuer might have its own crl or  oscp URL no?  We use the same caConfig intentionally right?

Reply:
it's `Issuer` has its own  crl and oscp URLs.
The CA config has default settings that can be applied for each issuers, if an issuer's settings are empty.
Usually CA will operate multiple issuers for different cert purposes, but the base URL can be the same, so no need to specify for each issuer. but if  needed, the settings should support it.